### PR TITLE
enhancement of data convertion at grpc channel

### PR DESCRIPTION
**Pull request by szc321**
Issue Number: 

Objective of pull request:

## Pull request checklist

Your PR fulfills the following requirements:
- [ ] [Issue](https://github.com/lava-nc/lava/issues) created that explains the change and why it's needed
- [ ] Tests are part of the PR (for bug fixes / features)
- [ ] [Docs](https://github.com/lava-nc/docs) reviewed and added / updated if needed (for bug fixes / features)
- [ ] PR conforms to [Coding Conventions](https://lava-nc.org/developer_guide.html#coding-conventions)
- [ ] [PR applys BSD 3-clause or LGPL2.1+ Licenses](https://lava-nc.org/developer_guide.html#add-a-license) to all code files
- [ ] Lint (`flakeheaven lint src/lava tests/`) and (`bandit -r src/lava/.`) pass locally
- [ ] Build tests (`pytest`) passes locally

## Pull request type

 

 

Please check your PR type:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes
- [ ] Other (please describe): 

## What is the current behavior?

-

## What is the new behavior?

-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by szc321** [comment]
the GrpcSendPortProxy can't inherit the SendPortProxy, because the send_port_  between the two Class is different.

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/port_proxy.cc [hunk 1]
```diff
@@ -90,47 +171,15 @@ int trick() {
...
-    metadata->dims[i] = dims[i];
-    metadata->strides[i] = strides[i]/element_size_in_bytes;
-    if (strides[i] % element_size_in_bytes != 0) {
-      LAVA_LOG_ERR("numpy array stride not a multiple of element bytes\n");
-    }
+DataPtr SendPortProxy::DataFromObject_(py::object* object) {
+#if defined(GRPC_CHANNEL)
+  if (channel_type_== ChannelType::RPCCHANNEL) {
+    return GrpcMDataFromObject_(object);
+  } else {
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/port_proxy.cc:178]: remove else.

File: src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt [hunk 2]
```diff
@@ -20,6 +20,7 @@ add_executable(
   $<$<BOOL:${DDS_CHANNEL}>:cpptest/test_ddschannel.cc>)
 
 
+
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt:23]: remove useless change

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/port_proxy.cc [hunk 3]
```diff
@@ -15,6 +15,87 @@ namespace message_infrastructure {
...
+  auto array = reinterpret_cast<PyArrayObject*> (obj);
+  if (!PyArray_ISWRITEABLE(array)) {
+    LAVA_LOG(LOG_LAYER, "The array is not writeable\n");
+  }
+  // var from numpy
+  int32_t ndim = PyArray_NDIM(array);
+  auto dims = PyArray_DIMS(array);
+  auto strides = PyArray_STRIDES(array);
+  void* data_ptr = PyArray_DATA(array);
+  // auto dtype = PyArray_Type(array);  // no work
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/port_proxy.cc:36]: remove useless code

File: src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc [hunk 4]
```diff
@@ -72,7 +72,7 @@ void dds_target_fn_a2_bound(int loop,
 
 TEST(TestDDSDelivery, DDSLoop) {
   MultiProcessing mp;
-  int loop = 100000;
+  int loop = 1000;
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc:75]: restore loop value

File: src/lava/magma/runtime/message_infrastructure/test/cpptest/test_grpc_delivery.cc [hunk 5]
```diff
@@ -148,6 +162,7 @@ TEST(TestGRPCChannel, GRPCLoop) {
   mp.Stop(true);
   if (result != expect_result) {
     LAVA_DUMP(LOG_UTTEST, "expect_result: %d\n", expect_result);
+    printf("result: %ld\n", result);
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/test/cpptest/test_grpc_delivery.cc:165]: useless?

File: src/lava/magma/runtime/message_infrastructure/test/test_all_delivery.py [hunk 6]
```diff
@@ -160,7 +160,7 @@ def prepare_data():
     return np.concatenate((arr2, arr1))
 
 
-class TestShmDelivery(unittest.TestCase):
+class TestDelivery(unittest.TestCase):
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/test/test_all_delivery.py:163]: need to change ut func name?

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel_proxy.cc [hunk 7]
```diff
@@ -40,7 +40,7 @@ GetRPCChannelProxy::GetRPCChannelProxy(const std::string &url,
   ChannelFactory &channel_factory = GetChannelFactory();
   channel_ = channel_factory.GetRPCChannel(url, port, src_name, dst_name, size);
   send_port_ = std::make_shared<SendPortProxy>(channel_type,
-                                               channel_->GetSendPort());
+                                                   channel_->GetSendPort());
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel_proxy.cc:43]: code style

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/grpc/grpc_port.cc [hunk 8]
```diff
@@ -10,21 +10,6 @@ namespace message_infrastructure {
 
 namespace {
 
-GrpcMetaData MetaData2GrpcMetaData(MetaDataPtr metadata) {
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/grpc/grpc_port.cc:13]: creare an inline func at grpc_port.h?
we could call this func at cpp ut.

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel_proxy.cc [hunk 9]
```diff
@@ -50,7 +50,7 @@ GetRPCChannelProxy::GetRPCChannelProxy(const std::string &src_name,
   ChannelFactory &channel_factory = GetChannelFactory();
   channel_ = channel_factory.GetDefRPCChannel(src_name, dst_name, size);
   send_port_ = std::make_shared<SendPortProxy>(channel_type,
-                                               channel_->GetSendPort());
+                                                   channel_->GetSendPort());
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel_proxy.cc:53]: code style

**Review by killight98 — CHANGES_REQUESTED** [review]

File: src/lava/magma/runtime/message_infrastructure/test/test_all_delivery.py (hunk 6 above)
Comment (szc321) [reply src/lava/magma/runtime/message_infrastructure/test/test_all_delivery.py:163]: change to TestAllDelivery

**Review by killight98 — APPROVED** [review]

**Pull request merged**